streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/HighAvailabilityTaskAssignor.java

```diff
@@ -51,12 +57,12 @@ public boolean assign(final Map<UUID, ClientState> clients,
         final SortedSet<TaskId> statefulTasks = new TreeSet<>(statefulTaskIds);
         final TreeMap<UUID, ClientState> clientStates = new TreeMap<>(clients);
 
-        assignActiveStatefulTasks(clientStates, statefulTasks);
+        final Map<TaskId, UUID> statefulTasksClientMappings = assignActiveStatefulTasks(clientStates, statefulTasks);
 
         assignStandbyReplicaTasks(
             clientStates,
-            statefulTasks,
-            configs.numStandbyReplicas
+            statefulTasksClientMappings,
+            configs
         );
```

Comment:
I was wondering whether we can simply standby assignment if `configs.numStandbyReplicas == 0`. Here or as first step in the method body of  `assignStandbyReplicaTasks()`. In this way we can remove `NoopStandbyTaskAssignor`.

Reply:
Sure! done.
Personal preference. Having all the strategies of standby task assignment implementations in a single class makes unit testing a bit easier. But I do agree that removing one extra class is indeed good idea.